Смоделировать игру против лотерейного автомата типа “777”. Игрок платит 1 руб., после чего выпадает случайное целое число, равномерно распределенное от 0 до 999. При некоторых значениях числа игрок получает выигрыш.

777: 200 руб.
999: 100 руб.
555: 50 руб.
333: 15 руб.
111: 10 руб.
*77: 5 руб.
**7: 3 руб.
*00: 2 руб.
**0: 1 руб.

Вопросы:

Выгодна ли игра игроку?
Сколько в среднем приобретает или теряет игрок за одну игру?

In [3]:
import numpy as np

In [4]:
trials = 1000000    # Количество попыток
successes = 0    # Количество успехов

for i in range(trials):
    random_score = np.random.randint(0, 1000)    
    if random_score == 777:
        successes += 200
    elif random_score == 999:
        successes += 100
    elif random_score == 555:
        successes += 50
    elif random_score == 333:
        successes += 15
    elif random_score == 111:
        successes += 10
    elif random_score % 100 == 77:
        successes += 5
    elif random_score % 10 == 7:
        successes += 3
    elif random_score % 10 == 0:
        successes += 1
    elif random_score % 100 == 0:
        successes += 2
m_monte_carlo = successes / trials    # Вероятность из моделирования
m = 200 / 1000 + 100 / 1000 + 50 / 1000 + 15 / 1000 + 10 / 1000 + 5*9 / 1000 + 3*90 / 1000 + 2 / 100 + 1 / 10                       # Теоретическая вероятность

print(successes, trials, m_monte_carlo, m)

797687 1000000 0.797687 0.81


Игра не выгодна для игрока. В среднем игрок приобретает 0.81